assets/src/editor/events/hooks/use-editor-event-entity.js

```diff
@@ -0,0 +1,32 @@
+import { useSelect } from '@wordpress/data';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+/**
+ * A hook for retrieving the existing event stored in state (or returning
+ * a passed in value that is already an event entity.
+ *
+ * Why a passed through value?  There may be cases where this hook is used in
+ * a component where `eventEntity` is in the props so there's no need to retrieve
+ * from the store state. One of the rules of react hooks is that they must always
+ * be in the same order and execute.  So that means you can't put a hook inside
+ * a conditional.
+ *
+ * @param {BaseEntity|null} existingEventEntity  Optionally an existing event
+ * entity that will be returned if valid.
+ * @return {BaseEntity|null}  An instance of the Event or null if not available
+ */
+const useEditorEventEntity = ( existingEventEntity = null ) => {
+	const eventEntity = useSelect( ( select ) => {
+		// if already have an event entity, then return it.
+		if ( isModelEntityOfModel( existingEventEntity, 'event' ) ) {
+			return existingEventEntity;
+		}
+		const events = select( 'eventespresso/core' ).getEvents();
+		return events.length > 0 ?
+			events.slice( -1 ).pop() :
```

Comment:
this is copying the last element of the `events` array into a new array then popping the last element off...

why not just `events.pop()` ? ie: is there a need to preserve the `events` array here?

Reply:
I think I had that originally but the tests exposed issues with mutating the returned array (as `events.pop()` would actually _change_ the length of the original array).  Hence switching to doing this.

Reply:
why are the tests concerned about the length of the array or whether it is mutated?

like I said above, is there a need to preserve the events array here?

Looking at the full code for this function, it appears that this would only run once for any individual event, meaning the events array is basically just an unimportant stepping stone and can be disposed of once a `venueEntity` has been retrieved.

Reply:
> why are the tests concerned about the length of the array or whether it is mutated?

I'm not just talking about automated tests.  I'm talking about the user land testing I did as well.  The selector for getEvents is memoized which means it returns the same array for events if the state has not changed.  Thus if  you mutate that array, future calls to the selector will return the mutated value.  I didn't realize that until I tested and discovered that and thus made this change.

>  it appears that this would only run once for any individual event, meaning the events array is basically just an unimportant stepping stone and can be disposed of once a venueEntity has been retrieved.

I'm not sure where that reasoning comes from.  This is a hook that could be implemented anywhere, so the assumption that it would only run once is incorrect - it depends on how and where its implemented.